Flaky test io.prestosql.operator.TestTopNRowNumberOperator

On multiple builds I've notices crashing builds:

```
2020-01-23T23:43:01.8186049Z [ERROR] Crashed tests:
2020-01-23T23:43:01.8186986Z [ERROR] io.prestosql.operator.TestTopNRowNumberOperator
2020-01-23T23:43:01.8191041Z [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
2020-01-23T23:43:01.8203226Z [ERROR] Command was /bin/sh -c cd /home/runner/work/presto/presto/presto-main && /opt/hostedtoolcache/jdk/8.0.242/x64/jre/bin/java -Dfile.encoding=UTF-8 -Xmx2g -Xms2g -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError -XX:-OmitStackTraceInFastThrow -jar /home/runner/work/presto/presto/presto-main/target/surefire/surefirebooter4819993699408889106.jar /home/runner/work/presto/presto/presto-main/target/surefire 2020-01-23T23-27-41_879-jvmRun1 surefire1536696459845097091tmp surefire_06355875406572171182tmp
2020-01-23T23:43:01.8203614Z [ERROR] Error occurred in starting fork, check output in log
2020-01-23T23:43:01.8203855Z [ERROR] Process Exit Code: 3
2020-01-23T23:43:01.8204101Z [ERROR] Crashed tests:
2020-01-23T23:43:01.8204451Z [ERROR] io.prestosql.operator.TestTopNRowNumberOperator
2020-01-23T23:43:01.8205857Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
2020-01-23T23:43:01.8207504Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
2020-01-23T23:43:01.8208795Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
2020-01-23T23:43:01.8210312Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
2020-01-23T23:43:01.8212211Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
2020-01-23T23:43:01.8214330Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
2020-01-23T23:43:01.8216464Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
2020-01-23T23:43:01.8218502Z [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
2020-01-23T23:43:01.8220622Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
2020-01-23T23:43:01.8222657Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
2020-01-23T23:43:01.8224608Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
2020-01-23T23:43:01.8227308Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
2020-01-23T23:43:01.8229829Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
2020-01-23T23:43:01.8232648Z [ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
2020-01-23T23:43:01.8234639Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
2020-01-23T23:43:01.8235980Z [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
2020-01-23T23:43:01.8237355Z [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
2020-01-23T23:43:01.8238881Z [ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
2020-01-23T23:43:01.8240330Z [ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
2020-01-23T23:43:01.8241489Z [ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
2020-01-23T23:43:01.8242657Z [ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
2020-01-23T23:43:01.8243933Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-01-23T23:43:01.8252120Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-01-23T23:43:01.8253856Z [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-01-23T23:43:01.8254867Z [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-01-23T23:43:01.8257252Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
2020-01-23T23:43:01.8259087Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
2020-01-23T23:43:01.8261208Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
2020-01-23T23:43:01.8263078Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
2020-01-23T23:43:01.8264529Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-01-23T23:43:01.8266464Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-01-23T23:43:01.8268443Z [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-01-23T23:43:01.8269582Z [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-01-23T23:43:01.8271438Z [ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
2020-01-23T23:43:01.8273446Z [ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
2020-01-23T23:43:01.8275873Z [ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
```

--- a/presto-main/src/test/java/io/prestosql/operator/TestTopNRowNumberOperator.java
+++ b/presto-main/src/test/java/io/prestosql/operator/TestTopNRowNumberOperator.java
@@ -193,7 +193,7 @@ public class TestTopNRowNumberOperator
     public void testMemoryReservationYield()
     {
         Type type = BIGINT;
-        List<Page> input = createPagesWithDistinctHashKeys(type, 6_000, 600);
+        List<Page> input = createPagesWithDistinctHashKeys(type, 1_000, 500);
 
         OperatorFactory operatorFactory = new TopNRowNumberOperatorFactory(
                 0,
@@ -228,6 +228,6 @@ public class TestTopNRowNumberOperator
                 count++;
             }
         }
-        assertEquals(count, 6_000 * 600);
+        assertEquals(count, 1_000 * 500);
     }
 }

